omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/DiagnosisResource2_2.java

```diff
@@ -0,0 +1,217 @@
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DataDelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link Resource} for Diagnosis, supporting standard CRUD operations
+ */
+@Resource(name = RestConstants.VERSION_1 + "/patientdiagnoses", supportedClass = Diagnosis.class, supportedOpenmrsVersions = "2.2.*")
+public class DiagnosisResource2_2 extends DataDelegatingCrudResource<Diagnosis> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Diagnosis getByUniqueId(String diagnosisUuid) {
+		return Context.getDiagnosisService().getDiagnosisByUuid(diagnosisUuid);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#delete(java.lang.Object,
+	 *      java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	protected void delete(Diagnosis diagnosis, String reason, RequestContext requestContext) throws ResponseException {
+		Context.getDiagnosisService().voidDiagnosis(diagnosis, reason);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#newDelegate()
+	 */
+	@Override
+	public Diagnosis newDelegate() {
+		return new Diagnosis();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#save(java.lang.Object)
+	 */
+	@Override
+	public Diagnosis save(Diagnosis diagnosis) {
+		return Context.getDiagnosisService().save(diagnosis);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#purge(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void purge(Diagnosis diagnosis, RequestContext requestContext) throws ResponseException {
+		Context.getDiagnosisService().purgeDiagnosis(diagnosis);
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getRepresentationDescription(org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation representation) {
+		if (representation instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
```

Comment:
For the default representation, can you add Representation.REF for diagnosis, condition, and encounter?